source/task_scheduler.cpp

```diff
@@ -333,7 +333,9 @@ int TaskScheduler::Init(TaskSchedulerInitOptions options) {
 	}
 
 	// Set the properties for the current thread
+	// on linux this locks code to single core
```

Comment:
This *should* be just for the main thread. Since we use pthread_self()

```
bool SetCurrentThreadAffinity(size_t coreAffinity) {
	// TODO: OSX and MinGW Thread Affinity
#	if defined(FTL_OS_LINUX)
	cpu_set_t cpuSet;
	CPU_ZERO(&cpuSet);
	CPU_SET(coreAffinity, &cpuSet);

	int ret = pthread_setaffinity_np(pthread_self(), sizeof(cpu_set_t), &cpuSet);
	if (ret != 0) {
		// Clang-tidy can't figure out the #ifdefs
		// NOLINTNEXTLINE(readability-simplify-boolean-expr)
		return false;
	}
#	else
	(void)coreAffinity;
#	endif

	return true;
}
```

I agree we should set the thread affinities for the worker threads. But IMO, this comment isn't correct. Thoughts?